testcases/misc/math/float/main.c

```diff
@@ -44,47 +44,47 @@ void myexit(int x)
 	tst_exit();
 }
 
-TH_DATA *pcom;
-TH_DATA **tabcom;
-TH_DATA **tabcour;
+static TH_DATA *pcom;
+static TH_DATA **tabcom;
+static TH_DATA **tabcour;
 #ifndef	PATH_MAX
 #define PATH_MAX		1024
 #endif
-char datadir[PATH_MAX];		/* DATA directory */
+static char datadir[PATH_MAX];		/* DATA directory */
 
 #ifndef PTHREAD_THREADS_MAX
 #define PTHREAD_THREADS_MAX	1024
 #endif
 #define DEFAULT_NUM_THREADS	20
-int num_threads = DEFAULT_NUM_THREADS;
-int num_loops = 500;
+static int num_threads = DEFAULT_NUM_THREADS;
+static int num_loops = 500;
 
-int sig_cancel = 0;		/* flag set by handle_signals to tell initial thread
+static int sig_cancel = 0;		/* flag set by handle_signals to tell initial thread
 				   to stop creating new threads (signal caught) */
 
-int indice = 0;			/* # of threads created, to be canceled by handle_signals
+static int indice = 0;			/* # of threads created, to be canceled by handle_signals
 				   or waited for by initial thread */
 
-pthread_mutex_t sig_mutex;
-pthread_t *threads;
+static pthread_mutex_t sig_mutex;
+static pthread_t *threads;
 
-int debug = 0;
-int true = 1;
+static int debug = 0;
+static int is_true = 1;
 
 static void *handle_signals(void *);
 
 static void sys_error(const char *, int);
 
-const double EPS = 0.1e-300;
+static const double EPS = 0.1e-300;
 
-const int nb_func = NB_FUNC;
+static const int nb_func = NB_FUNC;
 
-int generate(char *datadir, char *bin_path)
+static int generate(char *datadir, char *bin_path)
 {
 	char *cmdline;
-	char *fmt = "cd %s; %s/%s %s";
+	const char *fmt = "cd %s; %s/%s %s";
 
-	cmdline = malloc(2 * strlen(bin_path) + strlen(datadir) + strlen(GENERATOR) + strlen(fmt));
+	cmdline = (char *)malloc(2 * strlen(bin_path) + strlen(datadir) + strlen(GENERATOR) + strlen(fmt));
 	if (cmdline == NULL)
```

Comment:
If I remember @metan-ucw correctly, casting to `malloc()` is not required.

Also if it's not too hard for you, could you send patches to mailing list? https://github.com/linux-test-project/ltp/wiki#developers-corner

Reply:
without cast compiler can generate error:
`error: invalid conversion from ‘void*’ to ‘<some type>*’ [-fpermissive]`

Reply:
If that is a case either you are trying to compile C code with C++ compiler or your compiler is broken.

Reply:
That is true. I am using C++.

Reply:
Compiling LTP with C++ compiler is not supported as C++ is subtly incompatible with C.

Reply:
Ok. I have reverted type casts.